doc/rst/language/spec/variables.rst

```diff
@@ -229,39 +267,97 @@ initialization, not assignment, of that variable.
       5
 
 
-Split initialization does not apply:
-
- * when the variable is a field, config variable, or ``extern`` variable.
- * when an applicable assignment statement setting the variable could not
-   be identified
- * when an applicable assignment statement is in one branch of a
-   conditional but not in the other, unless:
+   *Example (split-init-select-dce.chpl)*
 
-     * the variable is not an ``out`` intent formal, and
-     * the other branch always returns or throws.
+   When the condition for a ``select`` or conditional statement
+   can be evaluated at compile-time, the compiler only considers
+   the path that will be taken:
+    
+   .. code-block:: chapel
+      
+      proc splitInitsBecauseDCE(type T) {
+        var x: int;
+        select T {
+          when int {
+            // split initialization occurs here when T==int
+            x = 1; 
+          }
+          when string {
+            writeln("compiler ignores this block when T==int");
```

Comment:
this feels more like a comment. If you want it to demonstrate it, you could have a comment and a `writeln` both:

```
// compiler ignores this block when T==int
writeln("T==string");
```